weave/trace_server/opentelemetry/python_spans.py

```diff
@@ -0,0 +1,342 @@
+    dropped_links_count: int = 0
+    status: Status = field(default_factory=Status)
+
+    @property
+    def start_time(self) -> datetime.datetime:
+        """Return the start time as a datetime object."""
+        return datetime.datetime.fromtimestamp(
+            self.start_time_unix_nano / 1_000_000_000
+        )
+
+    @property
+    def end_time(self) -> datetime.datetime:
+        """Return the end time as a datetime object."""
+        return datetime.datetime.fromtimestamp(self.end_time_unix_nano / 1_000_000_000)
+
+    @property
+    def duration_ns(self) -> int:
+        """Return the duration in nanoseconds."""
+        return self.end_time_unix_nano - self.start_time_unix_nano
+
+    @property
+    def duration_ms(self) -> float:
+        """Return the duration in milliseconds."""
+        return self.duration_ns / 1_000_000
+
+    @classmethod
+    def from_proto(cls, proto_span: PbSpan) -> "Span":
+        """Create a Span from a protobuf Span."""
+        return cls(
+            name=proto_span.name,
+            trace_id=hexlify(proto_span.trace_id).decode("ascii"),
+            span_id=hexlify(proto_span.span_id).decode("ascii"),
+            start_time_unix_nano=proto_span.start_time_unix_nano,
+            end_time_unix_nano=proto_span.end_time_unix_nano,
+            kind=SpanKind.from_proto(proto_span.kind),
+            parent_id=hexlify(proto_span.parent_span_id).decode("ascii")
+            if proto_span.parent_span_id
+            else None,
+            trace_state=proto_span.trace_state,
+            flags=proto_span.flags,
+            attributes=Attributes.from_proto(proto_span.attributes),
+            dropped_attributes_count=proto_span.dropped_attributes_count,
+            events=[Event.from_proto(e) for e in proto_span.events],
+            dropped_events_count=proto_span.dropped_events_count,
+            links=[Link.from_proto(l) for l in proto_span.links],
+            dropped_links_count=proto_span.dropped_links_count,
+            status=Status.from_proto(proto_span.status),
+        )
+
+    def to_call(
+        self, project_id: str
+    ) -> tuple[tsi.StartedCallSchemaForInsert, tsi.EndedCallSchemaForInsert]:
+        attributes = to_json_serializable(self.attributes._attributes)
+        # Options: set
+        start_call = tsi.StartedCallSchemaForInsert(
+            project_id=project_id,
+            id=self.span_id,
+            op_name=self.name,
+            trace_id=self.trace_id,
+            parent_id=self.parent_id,
+            started_at=self.start_time,
+            attributes=attributes,
+            inputs={},
+            wb_user_id=None,
+            wb_run_id=None,
+        )
+        # TODO: (Followup) Get Exception from Output
+
+        summary_insert_map = tsi.SummaryInsertMap(usage={})
+
+        end_call = tsi.EndedCallSchemaForInsert(
+            project_id=project_id,
+            id=self.span_id,
+            ended_at=self.end_time,
+            exception=None,
```

Comment:
should `exception` depend on status?

Reply:
I updated it to set the status and tested it out. The message doesn't seem to display properly in the dash unfortunately but it does when you click in.

![image](https://github.com/user-attachments/assets/648380e3-c6dc-4f29-bc63-4a278052e8f3)
![image](https://github.com/user-attachments/assets/dfa0638e-b58d-42c3-85a3-ef7cbee165a6)

Reply:
what is wrong with the dash?

Reply:
I thought that the lowercase status field would display the error message rather than just `error` but maybe that's not the case

Reply:
what is that lower case status? that looks like a custom field, not a standard field

Reply:
I fixed it, misread a schema, we can resolve this now